be/src/pipeline/task_scheduler.h

```diff
@@ -91,8 +91,7 @@ class TaskScheduler {
 
     void shutdown();
 
-    void update_tg_cpu_share(const taskgroup::TaskGroupInfo& task_group_info,
-                             taskgroup::TaskGroupPtr task_group);
+    TaskQueue* task_queue() const { return _task_queue.get(); }
```

Comment:
_task_queue is shared by task_scheduler and TaskGroup, so it' better use shared_ptr here.